app/demo-loop/LDemoRun.cc

```diff
@@ -128,48 +128,49 @@ LDemoResult run_gpu(LDemoArgs args)
            args.max_num_tracks);
     StateDeviceRef states_ref = make_ref(state_storage);
 
-    // Copy primaries to device and create track initializers
-    // TODO: for now this assumes we can initialize all primaries at once, but
-    // we should also handle the case where we have more primaries than trackss
```

Comment:
I know I wrote this comment, but looking at it again now I think this should actually say the case when the number of primaries is greater than the capacity of the track initializer storage. It's ok to have more primaries than track states (they'll be stored as track initializers along with any secondaries produced until there's an empty slot they can fill); it's if there weren't enough space to copy all the primaries to device at the start that we'd need this outer loop over primaries.

Reply:
Hmm, good point. So I guess there's no point of changing `extend_from_primaries` then. For the CPU demo loop, I can get away with having max_num_tracks = 1 and a sufficiently large storage_factor. Now this is making me rethink whether we really want to add this outer loop.

Reply:
Agreed, I think we can just make sure the storage factor is large enough instead of trying to handle this case, and since you're using the same track initialization logic on the CPU side same thing there. Sorry for the confusion!

Reply:
Ok, abort mission!